nodebb

```diff
@@ -2,36 +2,52 @@
 
 'use strict';
 
-var cproc;
-var args;
-var fs;
-var path;
+var fs = require('fs');
+var path = require('path');
+var cproc = require('child_process');
+
+// check to make sure dependencies are installed
+try {
+	fs.readFileSync(path.join(__dirname, 'node_modules/async/package.json'));
+} catch (e) {
+	if (e.code === 'ENOENT') {
+		process.stdout.write('Dependencies not yet installed.\n');
+		process.stdout.write('Installing them now...\n\n');
+
+		cproc.execSync('npm i --production', {
+			cwd: __dirname,
+			stdio: [0, 1, 2],
+		});
+	} else {
+		throw e;
+	}
+}
+
+var minimist;
 var request;
 var semver;
 var prompt;
 var async;
 
 try {
 	require('colors');
-	cproc = require('child_process');
-	args = require('minimist')(process.argv.slice(2));
-	fs = require('fs');
-	path = require('path');
+	minimist = require('minimist');
 	request = require('request');
 	semver = require('semver');
 	prompt = require('prompt');
 	async = require('async');
 } catch (e) {
-	if (e.code === 'MODULE_NOT_FOUND') {
-		process.stdout.write('NodeBB could not be started because it\'s dependencies have not been installed.\n');
-		process.stdout.write('Please ensure that you have executed "npm install --production" prior to running NodeBB.\n\n');
-		process.stdout.write('For more information, please see: https://docs.nodebb.org/en/latest/installing/os.html\n\n');
-		process.stdout.write('Could not start: ' + e.code + '\n');
+	process.stdout.write(
+		'\x1b[31mNodeBB could not be initialised because there was an error while loading dependencies.\n' +
+		'Please run "\x1b[33mnpm install --production\x1b[31m" and try again.\x1b[0m\n\n' +
```

Comment:
Oh, is this for colours? You can use `colors` module for that.

`'Please run "' + 'npm install --production'.yellow + ': and try again."`